src/tls/test-server.c

```diff
@@ -120,6 +120,42 @@ block_sigchld (void)
   g_assert_cmpint (sigaction (SIGCHLD, &child_action, NULL), ==, 0);
 }
 
+/* check if we have a client certificate for a given cgroup */
+static bool
+check_for_certfile (TestCase *tc,
+                    char **out_contents)
+{
+  g_autoptr(GDir) dir = g_dir_open (tc->runtime_dir, 0, NULL);
+  g_assert (dir != NULL);
+
+  const char *name;
+  while ((name = g_dir_read_name (dir)))
+    {
+      g_autofree char *filename = g_build_filename (tc->runtime_dir, name, NULL);
+
+      g_autofree char *contents = NULL;
+      g_autoptr(GError) error = NULL;
+      if (!g_file_get_contents (filename, &contents, NULL, &error))
+        {
+          /* files are flying around all the time: this might reasonably fail */
+          if (g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
```

Comment:
Do they? I don't understand how this can be anything but predictable in the unit tests -- each test gets its own temporary directory, and there is no paralellism. The only case that I can imagine is that this gets called if there is no file at all, but then this function already cleanly returns `false`.

Reply:
We loop, waiting for the file to disappear during teardown, for example.  In that case we'd possibly see the directory entry but be unable to open the file, if the deletion happens in between.